tsec-libsodium/src/test/scala/tsec/libsodium/PKEncryptionTest.scala

```diff
@@ -0,0 +1,126 @@
+package tsec.libsodium
+
+import cats.effect.IO
+import tsec.common._
+import tsec.libsodium.cipher.PlainText
+import tsec.libsodium.pk.encryption._
+
+class PKEncryptionTest extends SodiumSpec {
+
+  behavior of "Sodium Public Key Encryption"
+
+  it should "encrypt and decrypt properly - normal" in {
+    forAll { (s: String) =>
+      val program: IO[String] = for {
+        senderKP    <- CryptoBox.generateKeyPair[IO]
+        recipientKP <- CryptoBox.generateKeyPair[IO]
+        encrypt     <- CryptoBox.encrypt[IO](PlainText(s.utf8Bytes), recipientKP.pubKey, senderKP.privKey)
+        decrypt     <- CryptoBox.decrypt[IO](encrypt, senderKP.pubKey, recipientKP.privKey)
+      } yield decrypt.toUtf8String
+      program.unsafeRunSync() mustBe s
+    }
+  }
+
+  it should "not encrypt and decrypt properly an improper keypair" in {
+    forAll { (s: String) =>
+      val program: IO[String] = for {
+        senderKP    <- CryptoBox.generateKeyPair[IO]
+        recipientKP <- CryptoBox.generateKeyPair[IO]
+        wrongKP     <- CryptoBox.generateKeyPair[IO]
+        encrypt     <- CryptoBox.encrypt[IO](PlainText(s.utf8Bytes), recipientKP.pubKey, senderKP.privKey)
+        decrypt     <- CryptoBox.decrypt[IO](encrypt, senderKP.pubKey, wrongKP.privKey)
+      } yield decrypt.toUtf8String
+
+      program.attempt.unsafeRunSync() mustBe a[Left[Exception, _]]
```

Comment:
```scala
program.attempt.unsafeRunSync must matchPattern {
  case Left(err: Exception) => 
}
```

will actually check the exception on the left, FWIW.

Reply:
I had no idea, woo thanks a lot!